### PR TITLE
added _mm_storeu_si16 and _mm_storeu_si64

### PR DESCRIPTION
**Pull request by lassade**
From: https://github.com/rust-lang/rust/issues/62743

Edit:
Sorry I kind mess it up the pull request, the issue #882 is a duplicate

### CODE REVIEW

**Comment by Amanieu** [comment]
It seems that we are also missing `_mm_storeu_si32`.

**Comment by lassade** [comment]
Indeed it was, I just added it

File: crates/core_arch/src/x86/sse.rs [hunk 1]
```diff
@@ -3820,6 +3845,62 @@ mod tests {
         assert_eq!(vals[ofs + 4], 0.0);
     }
 
+    #[simd_test(enable = "sse")]
+    unsafe fn test_mm_storeu_si16() {
+        let mut vals = [0x0Fu8; 20];
+        // NOTE: Will endianess be a problem?
```
Comment (Amanieu) [inline crates/core_arch/src/x86/sse.rs:3851]: x86 is always little-endian.

File: crates/core_arch/src/x86/sse.rs [hunk 2]
```diff
@@ -1453,6 +1453,31 @@ pub unsafe fn _mm_storer_ps(p: *mut f32, a: __m128) {
...
 }
 
+/// Store 16-bit integer from the first element of `a` into memory.
+///
+/// `mem_addr` does not need to be aligned on any particular boundary.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_storeu_si16)
+#[inline]
+#[target_feature(enable = "sse")]
+//#[stable(feature = "simd_x86", since = "1.45.2")]
```
Comment (Amanieu) [inline crates/core_arch/src/x86/sse.rs:1463]: Please add stability attributes.

**Review by Amanieu — COMMENTED** [review]
Do a quick rustfmt run to fix the CI errors?

File: crates/core_arch/src/x86/sse.rs (hunk 2 above)
Comment (workingjubilee) [reply crates/core_arch/src/x86/sse.rs]: Unstable, a `feature` name like `"simd_x86_2"` is fine. It does not need a reason.

**Pull request closed without merging**